hooks/command

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+export DOCKER_CLI_EXPERIMENTAL=enabled
+export AWS_PAGER=""
+
+build="${BUILDKITE_BUILD_NUMBER:-0}"
+repository="$BUILDKITE_PLUGIN_DYNAMODB_IMAGE_REPOSITORY"
+tables=()
+script_dir="$(dirname "${BASH_SOURCE[0]}")"
+local_dynamo_container_id=""
+local_dynamo_port=8000
+tmp_dir="${script_dir}/tmp"
+mkdir -p $tmp_dir
+
+# Read a plugin property of type [array, string] into a Bash array. Buildkite
+# exposes a string value at BUILDKITE_PLUGIN_{NAME}_{KEY}, and array values at
+# BUILDKITE_PLUGIN_{NAME}_{KEY}_{IDX}.
+function read_list_property {
+  local prefix="BUILDKITE_PLUGIN_DYNAMODB_IMAGE_${1}"
+  local property="${prefix}_0"
+  result=()
+  if [[ -n ${!property:-} ]]; then
+    local i=0
+    local property="${prefix}_${i}"
+    while [[ -n ${!property:-} ]]; do
+      result+=("${!property}")
+      i=$((i + 1))
+      property="${prefix}_${i}"
+    done
+  elif [[ -n ${!prefix:-} ]]; then
+    result+=("${!prefix}")
+  fi
+  [[ ${#result[@]} -gt 0 ]] || return 1
+}
+
+# Read tables into an array
+function read_tables {
+  if read_list_property 'TABLES'; then
+    for table in "${result[@]}"; do
+      tables+=("${table}")
+    done
+  else
+    echo "A list of tables must be provided."
+  fi
+}
+
+# Retrieves schemas and saves them as JSON
+function retrieve_schemas {
+  for table in "${tables[@]}"; do
+    local schema_file="${tmp_dir}/${table}.json"
+    aws dynamodb describe-table \
+      --table-name "${table}" \
+      --output json \
+      >"${schema_file}"
+  done
+}
+
+# Pulls local DynamoDB and runs it on port 8000
+function start_local_dynamo {
+  docker pull amazon/dynamodb-local:latest
+  local_dynamo_container_id=$(docker run -d -p $local_dynamo_port:$local_dynamo_port amazon/dynamodb-local:latest -jar DynamoDBLocal.jar -port $local_dynamo_port -sharedDb)
+  sleep 5 # Give container a bit of time to start up
```

Comment:
Maybe we can add a TODO to more aggressively poll the container to see when it's up